subsys/spm/secure_services.c

```diff
@@ -47,6 +53,39 @@ int spm_secure_services_init(void)
 }
 
 
+#ifdef CONFIG_SPM_SERVICE_READ
+typedef struct {
+	u32_t start;
+	u32_t end;
+} read_range_t;
+
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_request_read(void *destination, const void *source, size_t num)
+{
+	static const read_range_t read_ranges[] = {
+#ifdef PM_MCUBOOT_ADDRESS
+		/* Allow reads of mcuboot metadata */
+		{.start = PM_MCUBOOT_PAD_ADDRESS,
+		 .end = PM_MCUBOOT_PAD_ADDRESS + PM_MCUBOOT_PAD_SIZE},
+#endif
+	};
+
+	if (destination == NULL || num <= 0) {
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < sizeof(read_ranges); i++) {
```

Comment:
sizeof(read_ranges) needs to be divided by sizeof(read_range_t).

Reply:
Will see if I can use ARRAY_SIZE instead